src/rgb20/wrapper.rs

```diff
@@ -251,6 +251,21 @@ impl Rgb20 {
             .sum()
     }
 
+    // Max supply for the inflation asset, if there is no `max supply`, then it will
+    // default to the non-inflatable asset `issued supply`
+    pub fn max_supply(&self) -> Amount {
+        self.0
+            .global("maxSupply")
+            .unwrap_or(
+                self.0
```

Comment:
We need this not to do an unnecessary secondary call to `global` if we have `maxSupply` in place
```suggestion
            .unwrap_or_else(|| self.0
```